pkgs/development/tools/build-managers/meson/default.nix

```diff
@@ -18,7 +18,11 @@ python3Packages.buildPythonApplication rec {
     popd
   '';
 
-  setupHook = ./setup-hook.sh;
+  postPatch = ''
+    sed -i -e 's|e.fix_rpath(install_rpath)||' mesonbuild/scripts/meson_install.py
```

Comment:
Can you add a comment why this necessary.

Reply:
@Mic92 Sure. If Meson is not patched, `result/bin/libinput` has empty rpath and consequently, it won't find `libinput.so` and other dependencies and won't run.

Reply:
Have you checked how if removing this is a good idea breaks other projects depending on this?

Reply:
I'm pretty sure nothing will break since most Meson-built derivations (there are currently only a few) patch ELF executables *after* compilation or set LD_LIBRARY_PATH:

https://github.com/NixOS/nixpkgs/blob/699231ed78ca41db175579bb1d7790a20beb75c4/pkgs/tools/filesystems/sshfs-fuse/default.nix#L33-L35
https://github.com/NixOS/nixpkgs/blob/35f89413e288f91837c625d38ac63eca9c2714b8/pkgs/os-specific/linux/firmware/fwupd/default.nix#L46-L49
https://github.com/NixOS/nixpkgs/blob/efeef044bef9cdcb7ee8431a8c174a7a20be93c7/pkgs/development/libraries/appstream-glib/default.nix#L31-L34

And one can't rely on rpath being empty, too. And we would be able to throw this fixup from other Meson derivations.

Reply:
What about users outside of nixpkgs. This is a significant derivation from upstream.

Reply:
What is the purpose of meson's behavior here? I would like to understand this because it might affect users using meson to build there own software outside of nix store.

Reply:
@mic92 it is purely a bug https://github.com/mesonbuild/meson/issues/314